command/CommandController.js

```diff
@@ -238,27 +253,38 @@ function scoutSummaries(getOptions, jsonBodyAttr, urlAttr, titleAttr, res) {
     });
 }
 
-function scoutTitles(getBody, res) {
+function scoutTitles(getBody, res, extendedData) {
   getOptions.body = JSON.stringify(getBody);
   rp(getOptions)
     .then(function(body) {
       const jsonBody = JSON.parse(body);
       if (jsonBody.status === 1 || jsonBody.status === 2) {
         console.log(jsonBody);
-        let articles = [];
+        let articlesPromises = [];
 
         // process list of articles
         Object.keys(jsonBody.list).forEach(key => {
           if (jsonBody.list[key].resolved_title) {
-            articles.push(getArticleMetadata(jsonBody.list[key]));
+            articlesPromises.push(
+              getArticleMetadata(jsonBody.list[key], extendedData)
+            );
           }
         });
 
-        articles.sort((a, b) => {
-          return a.sort_id - b.sort_id;
-        });
+        Promise.all(articlesPromises).then(function(values) {
+          faviconHelper.clearCurrentRequests();
```

Comment:
This call to faviconHelper here feels really out of context to the other things that are happening. From reading through the rest of this function, you would have no idea what this call was for or that you were expected to do some cleanup after calling `getArticleMetadata()`. I think the currentRequests activity needs to be handled inside of the FaviconHelper file.

Reply:
See comment on currentRequests[hostname]